Example/ScanDemo 2/Pods/CML_SCAN/ScanView.swift

```diff
@@ -0,0 +1,284 @@
+//
+//  ScanView.swift
+//  SCAN
+//
+//  Created by sukhjeet singh sandhu on 27/07/17.
+//  Copyright © 2017 Solcen. All rights reserved.
+//
+
+import UIKit
+import AVFoundation
+import AudioToolbox
+
+// These are the barcode types, those are supported by this Pod.
+public enum Types {
+    case qrCode
+    case dataMatrix
+    case ean8
+    case ean13
+    case code128
+    case code39
+}
+
+// This protocol is responsibe for providing the barcode and type back to the view controller.
+public protocol BarcodeScannerDelegate: class {
+    func didReceiveBarcode(barcode: String?, barcodeType: String?)
```

Comment:
This is very poor design. Optionals should not be used to express errors. If errors are a possibility, either return a `Result` enum with success and failure cases, or call different methods, one for success, and one for failure. The failure must always contain an `Error` confirming object.